src/chains/algorand_1/algoChainState.ts

```diff
@@ -183,4 +180,29 @@ export class AlgorandChainState {
     this.assertIsConnected()
     return this._algoClient
   }
+
+  // TODO: sort based on health info
+  /**  * Choose the best Chain endpoint based on health and response time */
+  private selectEndpoint(): { url: URL; endpoint: AlgorandChainEndpoint } {
+    // Just choose the first endpoint for now
+    const endpoint = this.endpoints[0]
+    const url = endpoint?.url?.href
+    return { url: new URL(trimTrailingChars(url, '/')), endpoint }
+  }
+
+  /** returns the required header from the array of objects. For ex: headers: [{'X-API-Key': '...'}]  */
+  private getHeader(key: string) {
```

Comment:
This should be more clear. I would rename to getHeaderValueFromActiveEndpoint()

Reply:
done. Also moved the getHeaderValueFromEndpoint function to generic helpers as other chains might need it too